scripts/deploy.js

```diff
@@ -30,24 +30,53 @@ async function main() {
         "FakeNFT": fakeNFT
     });
 
-    console.log("Minting 10 NFTs for each account...")
-    for (const account of accounts) {
-        await fakeNFT.connect(account).mint(10);
-        break;
-    }
+    const owner = accounts[0];
+    console.log(`Minting 10 NFTs for account address: ${owner.address}`)
+    await fakeNFT.connect(owner).mint(10);
 
-    owner = accounts[0];
-    console.log("Listing 5 NFTs from account " + owner.address);
+    console.log(`Listing 5 NFTs from account ${owner.address}`);
     await fakeNFT.setApprovalForAll(kasuContract.address, true);
-    
-    for (var i = 0; i < 5; i++) {
+
+    for (let i = 0; i < 5; i++) {
         id = await fakeNFT.tokenOfOwnerByIndex(owner.address, i);
-        await kasuContract.listNFT(id, fakeNFT.address, getRandomInt(1, 8), getRandomInt(1, 10), ethers.utils.parseEther("1").mul(getRandomInt(1, 1000)))
+        await kasuContract.listNFT(
+            id, // tokenId
+            fakeNFT.address, // tokenAddress
+            getRandomInt(1, 8), // duration
+            getRandomInt(1, 10), // dailyInterestRate
+            ethers.utils.parseEther("1").mul(getRandomInt(1, 10)) // collateralRequired in wei
+        );
     }
     events = await kasuContract.queryFilter("ListNFT");
+    const listingIds = [];
     for (const event of events) {
-        console.log(removeNumericKeys(event.args))
+        listingIds.push(removeNumericKeys(event.args).listingId);
     }
+    console.log(`Listing IDs: ${listingIds}`)
+
+    // Borrow
+    const account1 = accounts[1];
+    console.log(`Borrowing some NFTs from account ${account1.address}`)
+    const listings = await kasuContract.viewAllListings();
+    for (let i = 0; i < listings.length - 2; i++) {
+        const listing = listings[i];
+        const collateralRequiredInEth = Number(hre.ethers.utils.formatEther(listing.collateralRequired));
+        console.log(`Listing id: ${listing.id}, rent duration: ${listing.duration}, collateral req: ${collateralRequiredInEth} eth, interest rate: ${listing.dailyInterestRate}%`)
+        const totalAmountRequiredInEth =
+            collateralRequiredInEth  +
+            (collateralRequiredInEth * listing.duration * (listing.dailyInterestRate / 100));
```

Comment:
I'm getting an error
```
 Contract call:       Kasu#borrow
  From:                0x70997970c51812dc3a010c7d01b50e0d17dc79c8
  To:                  0x5fbdb2315678afecb367f032d93f642f64180aa3
  Value:               7.6999 ETH

  Error: VM Exception while processing transaction: reverted with reason string 'must send the exact payment'
```
I think you need to do the whole thing in BigNumber... see https://github.com/forcey/nft-rental-marketplace/blob/main/src/components/BorrowModal.tsx#L14, it's actually not very convoluted.

Reply:
@forcey ugh my bad. I tried to use this expression using bignumbers but got number underflow due to multiplying a value that's < 1. Didn't realize the expression you have also would work. Just pushed a fix for this.